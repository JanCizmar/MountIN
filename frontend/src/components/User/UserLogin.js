"use strict";

import React from 'react';
import {Link} from 'react-router-dom';

import {AlertMessage} from '../AlertMessage';
import Page from '../Page';
import {Button, Col, ControlLabel, FormControl, FormGroup, Row} from "react-bootstrap";


class UserLogin extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            username: '',
            password: ''
        };

        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleFormChange = this.handleFormChange.bind(this)

    }


    handleFormChange(event) {
        this.setState({
            [event.target.name]: event.target.value,
            changed: true
        });
    }

    handleSubmit(event) {
        event.preventDefault();

        let user = {
            username: this.state.username,
            password: this.state.password
        };

        this.props.onSubmit(user);
    }

    render() {
        return (
            <Page className="login-page">
                <Row>
                    <Col className="login-container" xs={10} md={4} sm={6} lg={4} xsOffset={1} mdOffset={4} smOffset={3}
                         lgOffset={4}>
                        <div className="login-heading">LOGIN</div>
                        <form onSubmit={this.handleSubmit}>
                            <FormGroup
                                controlId="username"
                                validationState={this.state.changed && this.state.username.length ? this.state.username.length < 5 || this.state.username.length > 200 ? 'error' : 'success' : null}
                            >
                                <ControlLabel>Username</ControlLabel>
                                <FormControl
                                    type="text"
                                    name="username"
                                    value={this.state.username}
                                    placeholder="Username"
                                    onChange={this.handleFormChange}
                                />
                            </FormGroup>
                            <FormGroup
                                controlId="password"
                                validationState={this.state.changed && this.state.password.length ? this.state.password.length < 5 || this.state.password.length > 200 ? 'error' : 'success' : null}
                            >
                                <ControlLabel>Password</ControlLabel>
                                <FormControl
                                    name="password"
                                    type="password"
                                    value={this.state.password}
                                    placeholder="Password"
                                    onChange={this.handleFormChange}
                                />
                            </FormGroup>

                            <Button className="login-button" id="submit" type="submit"
                                    disabled={this.state.username === undefined || this.state.username === '' || this.state.password === undefined || this.state.password === ''}
                            >Login</Button>

                            <AlertMessage
                                className="md-row md-full-width">{this.props.error ? `${this.props.error}` : ''}</AlertMessage>
                        </form>
                        <div className="register-link">
                            Not registered yet?<br/><Link to={'/register'} className=""> &nbsp;Click
                            Here&nbsp;</Link> to Register!
                        </div>
                    </Col>
                </Row>
            </Page>
        );
    }
}

export default UserLogin;