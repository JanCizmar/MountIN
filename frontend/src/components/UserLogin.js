"use strict";

import React from 'react';
import {Link, withRouter} from 'react-router-dom';

import {AlertMessage} from './AlertMessage';
import Page from './Page';
import {Button, Col, ControlLabel, FormControl, FormGroup} from "react-bootstrap";


class UserLogin extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            username: '',
            password: ''
        };

        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleFormChange = this.handleFormChange.bind(this)

    }


    handleFormChange(event) {
        this.setState({
            [event.target.name]: event.target.value
        });
    }

    handleSubmit(event) {
        event.preventDefault();

        let user = {
            username: this.state.username,
            password: this.state.password
        };

        this.props.onSubmit(user);
    }

    render() {
        return (
            <Page>
                <Row>
                    <Col lg={4} lgOffset={4}>
                        <form onSubmit={this.handleSubmit}>
                            <FormGroup
                                controlId="username"
                                validationState={this.state.username.length < 5 || this.state.username.length > 200 ? 'error' : 'success'}
                            >
                                <ControlLabel>Username</ControlLabel>
                                <FormControl
                                    type="text"
                                    name="username"
                                    value={this.state.username}
                                    placeholder="Username"
                                    onChange={this.handleFormChange}
                                />
                            </FormGroup>
                            <FormGroup
                                controlId="password"
                                validationState={this.state.password.length < 5 || this.state.password.length > 200 ? 'error' : 'success'}
                            >
                                <ControlLabel>Passowrd</ControlLabel>
                                <FormControl
                                    name="password"
                                    type="password"
                                    value={this.state.password}
                                    placeholder="Password"
                                    onChange={this.handleFormChange}
                                />
                            </FormGroup>

                            <Button id="submit" type="submit"
                                    disabled={this.state.username === undefined || this.state.username === '' || this.state.password === undefined || this.state.password === ''}
                            >Login</Button>
                            <Link to={'/register'} className="md-cell">Not registered yet?</Link>
                            <AlertMessage
                                className="md-row md-full-width">{this.props.error ? `${this.props.error}` : ''}</AlertMessage>
                        </form>
                    </Col>
                </Row>
            </Page>
        );
    }
}

export default withRouter(UserLogin);