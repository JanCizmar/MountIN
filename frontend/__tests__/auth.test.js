import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ReactRouterEnzymeContext from 'react-router-enzyme-context';
import toJson from 'enzyme-to-json';


import UserLogin from "../src/components/User/UserLogin";

configure({adapter: new Adapter()});

describe('<UserLogin />', () => {
    it('User login renders correctly', () => {
        const options = new ReactRouterEnzymeContext();
        const tree = shallow(
            <UserLogin/>,
            options.get());
        expect(toJson(tree)).toMatchSnapshot();
    });
});

describe('<UserSignup />', () => {
    it('User signup renders correctly', () => {
        const options = new ReactRouterEnzymeContext();
        const tree = shallow(
            <UserLogin/>,
            options.get());
        expect(toJson(tree)).toMatchSnapshot();
    });
});
