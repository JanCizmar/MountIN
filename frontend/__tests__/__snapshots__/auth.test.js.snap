// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserLogin /> User login renders correctly 1`] = `
<Page
  className="login-page"
>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      className="login-container"
      componentClass="div"
      lg={4}
      lgOffset={4}
      md={4}
      mdOffset={4}
      sm={6}
      smOffset={3}
      xs={10}
      xsOffset={1}
    >
      <div
        className="login-heading"
      >
        LOGIN
      </div>
      <form
        onSubmit={[Function]}
      >
        <FormGroup
          bsClass="form-group"
          controlId="username"
          validationState={null}
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Username
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            componentClass="input"
            name="username"
            onChange={[Function]}
            placeholder="Username"
            type="text"
            value=""
          />
        </FormGroup>
        <FormGroup
          bsClass="form-group"
          controlId="password"
          validationState={null}
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Password
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            componentClass="input"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
        </FormGroup>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className="login-button"
          disabled={true}
          id="submit"
          type="submit"
        >
          Login
        </Button>
        <styled.div
          className="md-row md-full-width"
        />
      </form>
      <div
        className="register-link"
      >
        Not registered yet?
        <br />
        <Link
          className=""
          replace={false}
          to="/register"
        >
            Click Here 
        </Link>
         to Register!
      </div>
    </Col>
  </Row>
</Page>
`;

exports[`<UserSignup /> User signup renders correctly 1`] = `
<Page
  className="login-page"
>
  <Row
    bsClass="row"
    componentClass="div"
  >
    <Col
      bsClass="col"
      className="login-container"
      componentClass="div"
      lg={4}
      lgOffset={4}
      md={4}
      mdOffset={4}
      sm={6}
      smOffset={3}
      xs={10}
      xsOffset={1}
    >
      <div
        className="login-heading"
      >
        LOGIN
      </div>
      <form
        onSubmit={[Function]}
      >
        <FormGroup
          bsClass="form-group"
          controlId="username"
          validationState={null}
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Username
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            componentClass="input"
            name="username"
            onChange={[Function]}
            placeholder="Username"
            type="text"
            value=""
          />
        </FormGroup>
        <FormGroup
          bsClass="form-group"
          controlId="password"
          validationState={null}
        >
          <ControlLabel
            bsClass="control-label"
            srOnly={false}
          >
            Password
          </ControlLabel>
          <FormControl
            bsClass="form-control"
            componentClass="input"
            name="password"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
          />
        </FormGroup>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className="login-button"
          disabled={true}
          id="submit"
          type="submit"
        >
          Login
        </Button>
        <styled.div
          className="md-row md-full-width"
        />
      </form>
      <div
        className="register-link"
      >
        Not registered yet?
        <br />
        <Link
          className=""
          replace={false}
          to="/register"
        >
            Click Here 
        </Link>
         to Register!
      </div>
    </Col>
  </Row>
</Page>
`;
